### PR TITLE
feat: documentation for AnalysisApps

### PR DESCRIPTION
**Pull request by asalzburger**
This adds documentation for the `ActsAnalysis*` applications, with example plots.

It also fixes a crash if run with `-h` option only, and fixes a wrong default tree name.

### CODE REVIEW

**Comment by asalzburger** [comment]
Hey @Corentin-Allaire - as discussed, here is the documentation update for the AnalysisApps.

File: docs/examples/analysis_apps.rst [hunk 1]
```diff
@@ -0,0 +1,230 @@
+ROOT based analysis applications
+================================
+
+The ACTS examples come with a certain variety of ROOT based validation and performance writers, 
+whose output can be use to understand various aspects of the reconstruction in more detail.
+
+The building of these applications can be switched on by setting ``ACTS_BUILD_ANALYSIS_APPS=On``,
+which requires (on top of the Core dependencies, also ROOT for the analysis code).
```
Comment (Corentin-Allaire) [inline docs/examples/analysis_apps.rst:8]: The parenthesis in that sentence are a bit weird.

**Review by Corentin-Allaire — COMMENTED** [review]
Great ! There is just one sentence that I find a bit weird.

File: docs/examples/analysis_apps.rst (hunk 1 above)
Comment (asalzburger) [reply docs/examples/analysis_apps.rst:8]: Fixed now.

**Review by Corentin-Allaire — APPROVED** [review]

**Pull request merged**